components/dashboard/src/prebuilds/list/ConfigurationInput.tsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { FC, useCallback, useMemo, useState } from "react";
+import { useConfiguration, useListConfigurations } from "../../data/configurations/configuration-queries";
+import { Combobox, ComboboxElement, ComboboxSelectedItem } from "@podkit/combobox/Combobox";
+
+interface RepositoryFinderProps {
+    selectedConfigurationId?: string;
+    disabled?: boolean;
+    expanded?: boolean;
+    onChange: (configurationId: string) => void;
+}
+
+export default function ConfigurationDropdown({
+    selectedConfigurationId,
+    disabled,
+    expanded,
+    onChange,
+}: RepositoryFinderProps) {
+    const [searchTerm, setSearchTerm] = useState("");
+    const { data, isLoading } = useListConfigurations({ searchTerm, sortBy: "creationTime", sortOrder: "desc" });
+    const configurations = data?.pages[0].configurations;
+    const { data: selectedConfiguration } = useConfiguration(selectedConfigurationId ?? ""); // we don't care if there is no data
+
+    const handleSelectionChange = useCallback(
+        (selectedId: string) => {
+            onChange(selectedId);
+        },
+        [onChange],
+    );
+
+    // Resolve the selected context url & project id props to a suggestion entry
+    const selectedSuggestion = useMemo(() => {
+        const match = configurations?.find((config) => {
+            return config.id === selectedConfigurationId;
+        });
+
+        return match;
+    }, [configurations, selectedConfigurationId]);
+
+    const getElements = useCallback(() => {
+        if (!configurations) {
+            return [];
```

Comment:
We can always keep `Show all` option

Reply:
Think we should go with https://github.com/gitpod-io/gitpod/pull/19354#discussion_r1473433773